src/invidious/helpers/oauth.cr

```diff
@@ -0,0 +1,41 @@
+require "oauth2"
+
+module Invidious::OAuthHelper
+  extend self
+
+  def get(key)
+    if HOST_URL == ""
+      raise Exception.new("Missing domain and port configuration")
+    end
+    if CONFIG.oauth[key]?
+      raise Exception.new("Invalid OAuth Endpoint: " + key)
+    end
+    v = CONFIG.oauth[key]
+    redirect_uri = HOST_URL + "/login/oauth/" + key
+
+    OAuth2::Client.new(v.host,
+      v.client_id, v.client_secret,
+      authorize_uri: v.auth_uri, token_uri: v.token_uri,
+      redirect_uri: redirect_uri)
```

Comment:
```suggestion
    if provider = CONFIG.oauth[key]?
      redirect_uri = "#{HOST_URL}/login/oauth/#{key}"

      OAuth2::Client.new(
        provider.host,
        provider.client_id,
        provider.client_secret,
        authorize_uri: provider.auth_uri,
        token_uri: provider.token_uri,
        redirect_uri: redirect_uri
      )
    else
      raise Exception.new("Invalid OAuth Endpoint: " + key)
    end
```

Reply:
The check on `CONFIG.oauth[key]?` is inverted, so when the provider was set in the config, it'd always raise an error. I changed that in the suggestion above, with some small style changes.

I've also used string interpolation (`#{}`) for `redirect_uri`. This is a minor performance change. For reference, see https://crystal-lang.org/reference/1.7/guides/performance.html#use-string-interpolation-instead-of-concatenation